utils/singularity.py

```diff
@@ -43,12 +47,20 @@ def run_in_tmp_dir():
     else:
         log.debug("Running in %s", running_in)
 
+    try:
+        _ = tempfile.mkdtemp(prefix=SCRATCH_NAME, dir=FWV0)
+        os.chdir(FWV0)  # run in /tmp/... directory so it is writeable
+        log.debug("Running in %s", FWV0)
+        return None
+    except OSError as e:
+        log.debug("Problem writing to %s: %s", FWV0, e.strerror)
+
     # This used to remove any previous runs (possibly left over from previous testing) but that would be bad
     # if other bids-fmripreps are running on shared hardware at the same time because their directories would
     # be deleted mid-run.  A very confusing error to debug!
 
     # Create temporary place to run gear
-    WD = tempfile.mkdtemp(prefix=SCRATCH_NAME, dir="/tmp")
+    WD = tempfile.mkdtemp(prefix=SCRATCH_NAME, dir=writable_dir)
```

Comment:
Not quite catching on to why tempfile.mkdtemp is called again, unless the above (L50-57) is just to check whether it can be created.

Reply:
Yes, that is to test if that directory is writable.

Changing that variable to /flywheel/v0 would be another way to do it.  My thought was to `cd` to the writable directory at this one point in the beginning of the code.   This means we have to find out the current directory in any code that `cd`s to some other directory so we can change back.  The alternative is to do as you suggest and then `cd` back there when necessary.